### PR TITLE
Issue #3817 countAndSay added

### PR DESCRIPTION
**Pull request by asimarise**
Issue #3817 is completed 

Added a new question for Strings Count and say

File is added in folder **NeoAlgo\C-Plus-Plus\cp** file name is **"countAndSay.cpp"**

A complete step-wise explanation with an example is provided with the code and also with the test cases and their expected output.

Please merge it and help me to make my first PR in GSSOC'21 journey.

### CODE REVIEW

File: C-Plus-Plus/cp/countAndSay.cpp [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+//Test case 1: 
+//Input : n = 6
+//Output : s = "312211"
+
+//Test case 2: 
+//Input : n = 9
+//Output : s = "31131211131221"
+
+//This Question has been asked in various MNC's coding round
```
Comment (HarshCasper) [inline C-Plus-Plus/cp/countAndSay.cpp:27]: Remove this

File: C-Plus-Plus/cp/countAndSay.cpp [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+//So for 4th(nth) term the sequence gives output of "1211"
+
+//Test case 1: 
+//Input : n = 6
+//Output : s = "312211"
+
+//Test case 2: 
+//Input : n = 9
+//Output : s = "31131211131221"
```
Comment (HarshCasper) [inline C-Plus-Plus/cp/countAndSay.cpp:25]: Put at the bottom of the code

File: C-Plus-Plus/cp/countAndSay.cpp [hunk 3]
```diff
@@ -0,0 +1,57 @@
+//Given a sequence of integer(input)
+// return the n'th term of the count-and-say sequence
+
+//The problem is to start count from the digit 1 and print(output) the counter and value of till that nth digit in the form of string
```
Comment (HarshCasper) [inline C-Plus-Plus/cp/countAndSay.cpp:4]: Use multi-line comments

File: C-Plus-Plus/cp/countAndSay.cpp [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+//Output: s = "1211"        where s is a string
+
+//Here we have to count till 4 and start from 1 digit so the our output value is now will be
+
+//step 1/(at 1st iteration) : count will "1" 
+// step 2                   : count "1" and say '"one" `1`' = "11"
+// step 3                   : count "11" and say '"two" `1`' = "21"
+// step 4/(nth iteration)   : count "21" and say '"one" `2` "one" `1`' = "12" + "11" = "1211"
+
+//So for 4th(nth) term the sequence gives output of "1211"
```
Comment (HarshCasper) [inline C-Plus-Plus/cp/countAndSay.cpp:17]: Not needed

File: C-Plus-Plus/cp/countAndSay.cpp [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+                    t = t + s[j-1];
+                    c = 1;
+                
+                } else c++;
+                
+            } s = t;
+            
+        } return s;
+    }
+};
```
Comment (HarshCasper) [inline C-Plus-Plus/cp/countAndSay.cpp:57]: Where is the driver code?

**Review by HarshCasper — CHANGES_REQUESTED** [review]

File: C-Plus-Plus/cp/countAndSay.cpp (hunk 5 above)
Comment (asimarise) [reply C-Plus-Plus/cp/countAndSay.cpp:57]: > Where is the driver code?

how to add that?

**Review by ankitaggarwal23 — COMMENTED** [review]
Code plagiarized from [here](https://www.geeksforgeeks.org/look-and-say-sequence/).

**Pull request closed without merging**